### PR TITLE
Normalize data

### PR DESCRIPTION
**Pull request by cserranocolome**
Included the file with the functions to normalize real data. 
I have also included this in the signature_net file (the one that runs everything) and I included the models, that were not included yet. Check this, please.

### CODE REVIEW

File: src/models/signature_net.py [hunk 1]
```diff
@@ -12,17 +20,35 @@ def __init__(self,
...
+        opp = create_opportunities(path_opportunities)
+        normalized_mutation_vec = normalize_data(mutation_vec, opp) 
+        normalized_mutation_vec = normalized_mutation_vec/np.sum(normalized_mutation_vec, axis=1)
+
+        # Run signature_finder
         weight_guess_0 = self.signature_finder.get_weights_batch(
             normalized_mutation_vec)
+
+        # Run finetuner
+        self.finetuner.load_state_dict(torch.load(os.path.join("../../trained_models", experiment_id_finetune), map_location=torch.device('cpu')))
```
Comment (OleguerCanal) [inline src/models/signature_net.py:45]: model path and name to lead should be not be hardcoded, could you do it an argument?

File: src/models/signature_net.py [hunk 2]
```diff
@@ -12,17 +20,35 @@ def __init__(self,
...
+        self.finetuner.eval()
         weight_guess_1 = self.finetuner(mutation_dist=normalized_mutation_vec,
                                            weights=weight_guess_0)
+
+        # Run error_finder
+        self.error_finder.load_state_dict(torch.load(os.path.join("../../trained_models", experiment_id_error_finder), map_location=torch.device('cpu')))
```
Comment (OleguerCanal) [inline src/models/signature_net.py:51]: same

File: src/models/signature_net.py [hunk 3]
```diff
@@ -12,17 +20,35 @@ def __init__(self,
...
         """Get weights of each signature in lexicographic wrt 1-mer
 
         Args:
             mutation_vec (np.array(batch_size, 96)): Batch of mutation vectors to decompose
+            path_opportunities (string): path to the txt file where the opportunities are. 
         """
+        # Normalize input data
         num_mutations = np.sum(mutation_vec, axis=1)
-        normalized_mutation_vec = mutation_vec/num_mutations
+        opp = create_opportunities(path_opportunities)
```
Comment (OleguerCanal) [inline src/models/signature_net.py:36]: this could be initialized in __init__, no need to reload it again each time we want to find the signature decomposition of a mutation

File: src/models/signature_net.py [hunk 4]
```diff
@@ -12,17 +20,35 @@ def __init__(self,
...
+        normalized_mutation_vec = normalize_data(mutation_vec, opp)
```
Comment (OleguerCanal) [inline src/models/signature_net.py:37]: nice :+1:

File: src/tests/model_tester.py [hunk 5]
```diff
@@ -66,39 +68,65 @@ def probs_batch_to_sigs(self, label_batch, predicted_batch, cutoff, num_classes)
...
+                              num_units_pos=num_neurons_pos,
+                              num_hidden_layers_neg=num_hidden_layers_neg,
+                              num_units_neg=num_neurons_neg,
+                              normalize_mut=normalize_mut)
+
+    model_error.load_state_dict(torch.load(os.path.join(
+        "../../trained_models", experiment_id_error_learner), map_location=torch.device('cpu')))
+    model_error.eval()
+    guessed_error_pos, guessed_error_neg = model_error(guessed_labels, num_mut)
+
```
Comment (OleguerCanal) [inline src/tests/model_tester.py:122]: I'm a bit confused by this:, isn't this what the class SignatureNet is supposed to be doing?

File: src/train_errorfinder.py [hunk 6]
```diff
@@ -41,7 +41,7 @@
                           num_hidden_layers=num_hidden_layers,
                           num_units=num_units)
     finetuner.to(device)
-    finetuner.load_state_dict(torch.load(os.path.join(model_path, finetuner_model_name)))
+    finetuner.load_state_dict(torch.load(os.path.join(model_path, finetuner_model_name), map_location=torch.device('cpu')))
```
Comment (OleguerCanal) [inline src/train_errorfinder.py:44]: Nice! :+1:

**Review by OleguerCanal — CHANGES_REQUESTED** [review]

File: src/models/signature_net.py (hunk 3 above)
Comment (cserranocolome) [reply src/models/signature_net.py:36]: I agree. Done

File: src/tests/model_tester.py (hunk 5 above)
Comment (cserranocolome) [reply src/tests/model_tester.py:122]: Yes. This is an outdated file that I reused to do a quick check. This is from the previous version.

Comment (OleguerCanal) [reply src/tests/model_tester.py:122]: :+1:

**Review by OleguerCanal — APPROVED** [review]
